src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -51,27 +51,36 @@ operator()(const NodeID, const EdgeID source_edge_id, Intersection intersection)
     if (intersection.size() <= 2)
         return intersection;
 
-    const auto findNextIntersectionForRoad =
-        [&](const NodeID at_node, const ConnectedRoad &road, NodeID &output_node) {
-            auto intersection = intersection_generator(at_node, road.turn.eid);
-            auto in_edge = road.turn.eid;
-            // skip over traffic lights
-            while (intersection.size() == 2)
+    const auto findNextIntersectionForRoad = [&](
+        const NodeID at_node, const ConnectedRoad &road, NodeID &output_node) {
+        auto intersection = intersection_generator(at_node, road.turn.eid);
+        auto in_edge = road.turn.eid;
+        // skip over traffic lights
+        // if there are more than 10 traffic lights/other items on the road, we don't consider it
+        // for sliproads anymore. This also prevents us from ending up in an endless loop.
```

Comment:
This is primarily for endless loops, right? We don't really care for number of traffic lights on sliproads?